src/trade-params/TradeParams.js

```diff
@@ -85,6 +85,10 @@ export default class TradeParams extends Component {
         this.onStopTypeChange = ::this.onStopTypeChange;
         this.onStopProfitChange = ::this.onStopProfitChange;
         this.onPurchase = ::this.onPurchase;
+
+        this.state = {
+            dynamicKey: 0,
+        };
```

Comment:
Controlled input does not work unless we introduce another middle state, because we want to allow user to input let say `12`, but by having it controlled, user can only input `1` or `2` but never `12`.

I think Introducing another middle state is worse than this approach.